backend/src/main/java/edu/alexu/cse/dripmeup/service/UserService.java

```diff
@@ -10,24 +12,25 @@
 import edu.alexu.cse.dripmeup.service.builder.UserPersonBuilder;
 
 @Service
+@RequiredArgsConstructor
 public class UserService {
 
     private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
 
-    @Autowired
-    public UserService(UserRepository userRepository) {
-        this.userRepository = userRepository;
-    }
 
     public boolean login(String email, String password) {
         UserEntity user = userRepository.findByEmail(email);
-        return user != null && user.getPassword().equals(password);
+//        System.out.println(user.getPassword());
+//        System.out.println(passwordEncoder.encode(password));
+        return user != null && passwordEncoder.matches(password, user.getPassword());
     }
 
     public Person signup(UserEntity user) throws HandlerException {
         if (userRepository.findByEmail(user.getEmail()) != null) {
             throw new HandlerException("Email already exists");
         }
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
         return new PersonDirector().construct(new UserPersonBuilder(user, userRepository));
     }
 
```

Comment:
In the **changePassword** method, the new password is directly saved without encoding it:
user.setPassword (newPassword.getPassword ()) ;